File: java/leetcode-practice/src/test/java/com/mamta/leetcode/twopointer/Leet167.java
package com.mamta.leetcode.twopointer;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
public class Leet167 {

    public int[] twoSum(int[] numbers, int target) {
        int left = 0, right = numbers.length - 1;

        while (left < right) {
            int sum = numbers[left] + numbers[right];

            if (sum == target) {
                return new int[]{left + 1, right + 1};
            } else if (sum < target) {
                left++;
            } else {
                right--;
            }
        }

        return new int[]{};
    }

    @Test
    public void testTwoSum() {
        int[] numbers = {2,7,11,15};
        int target = 9;
        int[] ans = twoSum(numbers, target);
        System.out.println(Arrays.toString(ans));
    }
}

File: python/leet167.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        numbers.sort()
        left = 0
        right = len(numbers)-1
        while left < right:
            sum = numbers[left]+numbers[right]
            if sum==target:
                return [left+1, right+1]
            if sum < target:
                left+=1
            else:
                right-=1



if __name__=="__main__":
    s = Solution()
    numbers = [2, 7, 11, 15]
    target = 9
    values = s.twoSum(numbers, target)
    print(values)